gateway-js/src/executeQueryPlan.ts

```diff
@@ -556,7 +556,7 @@ function doesTypeConditionMatch(
   }
 
   if (isAbstractType(conditionalType)) {
-    return schema.isSubType(conditionalType, type);
+    return schema.isSubType(conditionalType, type as any);
```

Comment:
It seems the types on `isSubType` are a bit more refined than before. We should be able to make some assertions about `type` before we make this call.

```
if (!isObjectType(type) || !isInterfaceType(type)) {
  return false;
}
```